packages/dart_frog_gen/lib/src/build_route_configuration.dart

```diff
@@ -225,9 +225,12 @@ List<RouteFile> _getRouteFiles({
     onRoute(route);
     files.add(route);
 
-    if (subDirectories.contains(path.basenameWithoutExtension(filePath))) {
-      onRogueRoute(route);
-    }
+    final fileBasename = path.basenameWithoutExtension(filePath);
+    final isRogueRoute = subDirectories.contains(fileBasename) &&
+        !File(path.join(directory.path, fileBasename, 'index.dart'))
```

Comment:
Nit: extract this `File` to a local variable for better reading.

Reply:
will update in a subsequent PR 👍